code/modules/food/recipes_microwave.dm

```diff
@@ -195,6 +180,36 @@ I said no!
 	)
 	result = /obj/item/chems/food/snacks/donkpocket //SPECIAL
 
+/decl/recipe/donkpocket/warm/check_items(obj/container) // NO FUCKING INFINITE LOOPS
```

Comment:
Seems vastly overcomplicated: can't you just check if there's a non-warm donk packet? You don't gain anything by making it more generic because as you say it's never going to be generic.

Reply:
That would stop you from cooking more than one donk pocket at a time unless I make it eject the results when they're made rather than after.

Reply:
Why? A warm donk packet doesn't prevent cooking, it's just not a valid ingredient. I don't get it.

Reply:
... I'm not sure what you're suggesting?